### ISSUE TITLE
Ignite fails to download kubectl and failed to create kubelet.service

### ISSUE THREAD

**Issue by abrarshivani** [kubernetes/kubernetes-anywhere#286]
When I built the docker container using DockerFile in phase2 and use that as ```.phase2.installer_container ``` , ignite fails to download kubectl and failed to create kubelet.service.
Here is output which I got while running kubernetes-anywhere,
```
null_resource.master (remote-exec): INFO     : files: createFilesystemsFiles: createFiles: op(11): op(12): [started]  GET "https://storage.googleapis.com/kubernetes-release/release/v1.3.4/bin/linux/amd64/kubectl"
null_resource.master (remote-exec): DEBUG    : files: createFilesystemsFiles: createFiles: op(11): op(12): GET result: OK
null_resource.master: Still creating... (59s elapsed)
null_resource.master (remote-exec): CRITICAL : files: createFilesystemsFiles: createFiles: op(11): op(12): [failed]   GET "https://storage.googleapis.com/kubernetes-release/release/v1.3.4/bin/linux/amd64/kubectl": net/http: request canceled (Client.Timeout exceeded while reading body)
null_resource.master (remote-exec): CRITICAL : files: createFilesystemsFiles: createFiles: op(11): [failed]   fetching file "/usr/local/bin/kubectl": net/http: request canceled (Client.Timeout exceeded while reading body)
null_resource.master (remote-exec): CRITICAL : files: failed to create files: failed to create files: failed to resolve file "/usr/local/bin/kubectl"
null_resource.master (remote-exec): Failed to execute operation: No such file or directory
null_resource.master (remote-exec): Failed to start kubelet.service: Unit kubelet.service failed to load: No such file or directory.
```

**Comment by colemickens** [comment]
I'm able to access that URL from a few networks I just tried.

Are you having network problems? ` GET "https://storage.googleapis.com/kubernetes-release/release/v1.3.4/bin/linux/amd64/kubectl": net/http: request canceled (Client.Timeout exceeded while reading body)`

**Comment by abrarshivani** [comment]
@colemickens No. When I download it manually inside the VMs it works.

**Comment by colemickens** [comment]
You've tested what multiple times? Running the whole deployment / ignition container? Or manually calling curl/wget in the VM? Or both?

What phase1 are you using?

**Comment by abrarshivani** [comment]
I have tested both. I am using ```vSphere Cloud Provider``` for phase1.

**Comment by colemickens** [comment]
I don't really know then.

Can you start a busybox one of the vsphere vms and try to manually download? Maybe something about docker's networking is not configured as needed for the vsphere phase1.

**Comment by colemickens** [comment]
I guess you contributed the vsphere phase1... Was it not working at one point in time?

**Comment by abrarshivani** [comment]
I tested with busybox and downloaded it manually, it worked fine. Yes, I contributed to vSphere phase1, it was not working at that time. So, I removed from ignite.json and build docker container ```docker.io/ashivani:v3 ``` and download it using curl in phase1.

**Comment by abrarshivani** [comment]
Did you find the same issue with Azure?

**Comment by colemickens** [comment]
No, can't say that I've seen it or had anyone report it. I'm not really sure what could be going on other than intermittent networking issues to be honest.

**Comment by abrarshivani** [comment]
@colemickens I think this is timeout issue for large files from ignition https://github.com/coreos/bugs/issues/1390. I tested this by adding small file to master.jsonnet and deployed kubernetes again, small file got download but large one failed. This is resolved [here](https://github.com/coreos/ignition/pull/255). But this change is not there in ignition version we use right now.

**Comment by colemickens** [comment]
I'd merge a PR that upgraded the version of ignition used.

**Comment by colemickens** [comment]
Especially if you validate that it fixes your problem.

**Comment by abrarshivani** [comment]
@colemickens I tested with updated version of ignition and it works fine. I think it solves this [issue](https://github.com/kubernetes/kubernetes-anywhere/pull/205) too. 

Now, to update iginition in kubernetes-anywhere, I can upload binary at some location and provide it in DockerFile as we do it now. Yet the ```vanilla``` oem is not upstream to coreos/ingnition it resides on @mikedanese fork. I believe that it is good idea to upstream onto main repo because it will be easy to maintain and update. For instance, to update the version I had to make lot of changes to the code since interface for provider is changed in coreos/ignition.

@mikedanese Can I upstream ```vanilla oem``` since I have recent changes?
